lib/core/src/Cardano/Wallet/Primitive/Types/Tx/Gen.hs

```diff
@@ -104,12 +105,12 @@ genTxInLargeRange = TxIn
 genTxOutSmallRange :: Gen TxOut
 genTxOutSmallRange = TxOut
     <$> genAddressSmallRange
-    <*> genTokenBundleSmallRange
+    <*> genTokenBundleSmallRange `suchThat` (/= TokenBundle.empty)
```

Comment:
Nice addition, very concise and clear. 👍🏻

Reply:
I was puzzled by this one test failing, though i had run the test suite already a few time. Drilling a bit more into the problem, I found this. I don't think we should ever have an empty token bundle except as a starting point for folding, though there might be cases where it makes sense :thinking: ? The TxOut is definitely not one of them IMO, so I've only banished them there (and not in the token bundle generator itself).

Reply:
> The TxOut is definitely not one of them IMO, so I've only banished them there (and not in the token bundle generator itself).

Yes, it's definitely the right decision to banish them here.

The token bundle generator itself still needs to generate `TokenBundle.empty` occasionally, since some of the compatibility properties rely on that to meet their coverage checks.

Reply:
@KtorZ I've just realized that this is not **_quite_** correct.

I believe the correct filter condition should be that **the ada value is non-zero**. This is because it's possible to have a non-empty token bundle with an ada quantity of zero, which we don't want to see in `TxOut`.

I've added a commit to my PR that makes this change [here](https://github.com/input-output-hk/cardano-wallet/pull/2450/commits/912ea634fd671985f5578979c3bd55518d997957), as it's necessary to make the property tests for `performSelection` pass.